File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Commons/UserLog.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Entity.Commons
{
    public class UserLog : DSWBaseEntity
    {
        #region [ Constructor ]

        public UserLog() : this(Guid.NewGuid()) { }
        public UserLog(Guid uniqueId)
            : base(uniqueId)
        {
        }
        #endregion

        #region[ Properties ]
        public string SystemUser { get; set; }
        public string SystemServer { get; set; }
        public string SystemComputer { get; set; }
        public int? AccessNumber { get; set; }
        public DateTimeOffset? PrevOperationDate { get; set; }
        public string SessionId { get; set; }
        public bool? AdvancedScanner { get; set; }
        public bool? AdvancedViewer { get; set; }
        public string UserMail { get; set; }
        public string MobilePhone { get; set; }
        public string DefaultAdaptiveSearchControls { get; set; }
        public string AdaptiveSearchStatistics { get; set; }
        public string AdaptiveSearchEvaluated { get; set; }
        public int PrivacyLevel { get; set; }
        public Guid? CurrentTenantId { get; set; }
        public string UserProfile { get; set; }
        #endregion

        #region [ Navigation Properties ]
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Protocol/Log/UserLog.vb
<Serializable()> _
Public Class UserLog
    Inherits AuditableDomainObject(Of String)

#Region "private data"

#End Region

#Region "Properties"
    Public Overridable Property SystemServer As String
    Public Overridable Property SystemComputer As String
    Public Overridable Property LastOperationDate As DateTimeOffset?
    Public Overridable Property AccessNumber As Integer
    Public Overridable Property PrevOperationDate As DateTimeOffset?
    Public Overridable Property SessionId As String
    Public Overridable Property AdvancedScanner As Integer
    Public Overridable Property AdvancedViewer As Integer
    Public Overridable Property CurrentTenantId As Guid
    Public Overridable Property UserMail As String
    Public Overridable Property MobilePhone As String
    Public Overridable Property DefaultAdaptiveSearchControls As String
    Public Overridable Property AdaptiveSearchStatistics As String
    Public Overridable Property AdaptiveSearchEvaluated As String
    Public Overridable Property PrivacyLevel As Integer
    Public Overridable Property UserProfile As String
    Public Overridable Property UserPrincipalName As String

#End Region

#Region "Ctor/init"
    Public Sub New()
        UniqueId = Guid.NewGuid()
    End Sub
#End Region
End Class


